gdzig/class/Object.mixin.zig

```diff
@@ -97,6 +97,52 @@ fn typeToken(comptime T: type) *anyopaque {
     }.token);
 }
 
+/// Connects a signal to a callable.
+pub fn connect(self: *Self, comptime S: type, callable: Callable) ConnectError!void {
+    const signal_name: StringName = .fromComptimeLatin1(comptime gdzig_meta.signalName(S));
+    const result = self.connectRaw(signal_name, callable, .{});
+    if (result != .ok) return ConnectError.AlreadyConnected;
+}
+
+/// Disconnects a signal from a callable.
+pub fn disconnect(self: *Self, comptime S: type, callable: Callable) void {
+    const signal_name: StringName = .fromComptimeLatin1(comptime gdzig_meta.signalName(S));
+    self.disconnectRaw(signal_name, callable);
+}
+
+/// Emits a signal.
+pub fn emit(self: *Self, signal: anytype) EmitError!void {
+    const S = @TypeOf(signal);
+    const signal_name: StringName = .fromComptimeLatin1(comptime gdzig_meta.signalName(S));
+    const fields = @typeInfo(S).@"struct".fields;
+    var args: std.meta.Tuple(&typeInfoToTypes(fields)) = undefined;
+    inline for (fields, 0..) |field, i| {
+        args[i] = @field(signal, field.name);
+    }
+    switch (self.emitRaw(signal_name, args)) {
+        .ok => {},
+        .err_unavailable => {
+            // Godot does not distinguish between "not a signal I handle" and "no one is listening to this signal"
```

Comment:
I don't think "no one is listening to this signal" is an error? It should just be a no-op.

Reply:
that's what this logic is saying